sklearn/metrics/cluster/tests/test_supervised.py

```diff
@@ -370,3 +379,70 @@ def test_check_clustering_error():
 
     with pytest.warns(UserWarning, match=msg):
         check_clusterings(wavelength, noise)
+
+
+def test_pair_confusion_matrix():
+    n = 10
+    N = n * n
+
+    # edge case: every element is its own cluster
+    clustering1 = list(range(N))
+    clustering2 = clustering1
+    expected = np.array([[N * (N - 1), 0], [0, 0]])
+    assert_array_equal(expected, pair_confusion_matrix(clustering1,
+                                                       clustering2))
+
+    # edge case: only one cluster
+    clustering1 = np.full((N,), fill_value=0)
+    clustering2 = clustering1
+    expected = np.array([[0, 0], [0, N * (N - 1)]])
+    assert_array_equal(expected, pair_confusion_matrix(clustering1,
+                                                       clustering2))
+
+    # regular case: different non-trivial clusterings
+    clustering1 = np.array([i+1 for i in range(n) for j in range(n)])
+    clustering2 = np.array([i+1 for i in range(n) for j in range(n+1)][:N])
+    # basic quadratic implementation
+    expected = np.full(shape=(2, 2), fill_value=0, dtype=np.int64)
+    for i in range(len(clustering1)):
+        for j in range(len(clustering2)):
+            if i != j:
+                same_cluster_1 = int(clustering1[i] == clustering1[j])
+                same_cluster_2 = int(clustering2[i] == clustering2[j])
+                expected[same_cluster_1, same_cluster_2] += 1
+    assert_array_equal(expected, pair_confusion_matrix(clustering1,
+                                                       clustering2))
```

Comment:
```suggestion
    assert_array_equal(
        expected, pair_confusion_matrix(clustering1, clustering2)
    )
```

Reply:
In general the numpy.testing API expects the arguments to be of the order (actual, desired, ...), so I changed this to:
```
    assert_array_equal(
        pair_confusion_matrix(clustering1, clustering2), expected)
    )
```
The same comment about the argument order applies to the other tests I implemented.

Reply:
Yes you are right. In testing the order is usually `assert computed == expected`.